lib/trino-parquet/src/main/java/io/trino/parquet/metadata/ParquetMetadata.java

```diff
@@ -106,9 +106,7 @@ public List<BlockMetadata> getBlocks(long splitStart, long splitLength)
         if (rowGroups != null) {
             for (RowGroup rowGroup : rowGroups) {
                 if (rowGroup.isSetFile_offset()) {
-                    long rowGroupStart = rowGroup.getFile_offset();
-                    boolean splitContainsRowGroup = splitStart <= rowGroupStart && rowGroupStart < splitStart + splitLength;
-                    if (!splitContainsRowGroup) {
+                    if (notOverLap(splitStart, splitLength, rowGroup)) {
```

Comment:
The original logic here was
```
long blockStart = block.getStartingPos();
boolean splitContainsBlock = splitStart <= blockStart && blockStart < splitStart + splitLength;
```
`blockStart` is supposed to be the same thing as `rowGroupStart`.
The intent is to include a row-group within a split if it starts within the boundary of the split. It doesn't matter if the row-group ends beyond the boundary of the split, it will be included within current split and skipped by the next one. This way every row-group is included once in some split.

Reply:
> blockStart is supposed to be the same thing as rowGroupStart.

Do you mean, the `blockStart` value belongs to a set of the `rowGroupStart` ?

Reply:
I meant that it is supposed to be exactly the same value (blockStart is starting position of first column in a row group). Do you see a situation where these values differ ?

Reply:
I put the code like this here:
```
        boolean splitContainsRowGroup = splitStart <= rowGroupStart && rowGroupStart < splitStart + splitLength;
        if (splitContainsRowGroup) {
            if (splitStart != rowGroupStart) {
                throw new RuntimeException("splitStart != rowGroupStart: " + splitStart + " != " + rowGroupStart);
            }
        }
```
It can not start when copying tpch table:`CREATE TABLE IF NOT EXISTS nation AS SELECT * FROM tpch.tiny.nation`

Reply:
and put this code in `PredicateUtils`:
```
           if (splitContainsBlock) { // line 202
                if (splitStart != blockStart) {
                    throw new RuntimeException("splitStart != rowGroupStart: " + splitStart + " != " + blockStart);
                }
```
Run the test, you will see 
```
Error opening Iceberg split local:///tpch/test_correctness_for_deletef1u2udien1-a1d9e56f61804f2aad6c114e04a91337/data/created_at_year=2024/20250228_071530_00022_qnuz6-ec1eab62-8476-47d3-b63b-cc317c042a75.parquet (offset=0, length=1855144): splitStart != rowGroupStart: 0 != 4
```

Reply:
I think I misunderstood your commit, I may need more time to read the code

Reply:
The problem is that once the rowGroup is skiped, there is no chance to update the `fileRowCount` outside, in `PredicateUtils` it will be kept in `RowGroupInfo`.

Reply:
ya, I meant that blockStart is supposed to match rowGroupStart. It's fine for splitStart to be different, that's expected.

Reply:
> The problem is that once the rowGroup is skiped, there is no chance to update the fileRowCount outside, in PredicateUtils it will be kept in RowGroupInfo.

That is indeed a bug. I think we can resolve that by adding `fileRowOffset` to `BlockMetadata` in `io.trino.parquet.metadata.ParquetMetadata#getBlocks(long, long)`